src/compiler/crystal/macros/methods.cr

```diff
@@ -1905,12 +1905,15 @@ module Crystal
     end
 
     def self.has_constant?(type, name)
-      BoolLiteral.new(type.types.has_key?(name))
+      path = Path.new name.split("::", remove_empty: true), name.starts_with?("::")
```

Comment:
Now this would allow things such as `::::Compress::::::Zip::::Reader` to succeed. Invalid syntax should not be allowed here.

Reply:
It might be easier at this point to just use `Parser#parse_path`. This should not be hard to setup and ensures exactly the same behaviour as a literal Path.

Reply:
Thanks, I'll see about switching things over after work today.